lib/identity_cache/cached/primary_index.rb

```diff
@@ -57,13 +57,24 @@ def load_one_from_db(id)
         record
       end
 
-      def load_multi_from_db(ids)
-        return {} if ids.empty?
-
-        ids = ids.map { |id| model.connection.type_cast(id, id_column) }
-        records = build_query(ids).to_a
-        model.send(:setup_embedded_associations_on_miss, records)
-        records.index_by(&:id)
+      if ActiveRecord.gem_version >= Gem::Version.new('6.1')
+        def load_multi_from_db(ids)
+          return {} if ids.empty?
+
+          ids = ids.map { |id| model.connection.type_cast(id) }
```

Comment:
Do you happen to know why this line is needed in the first place?  It seems like `build_query` on the following line should already be doing casting through `model.where(model.primary_key => id_or_ids)`.

Reply:
That's a good question, I've been a bit head down and haven't paid too much attention. I'll try it out and see.